### PR TITLE
Feature/faster transfer

### PR DESCRIPTION
**Pull request by gregorjerse**

### CODE REVIEW

File: resolwe/flow/executors/init_container.py [hunk 1]
```diff
@@ -65,19 +67,36 @@ async def transfer_inputs(communicator: BaseCommunicator):
...
+    # Group files by connectors. So we can transfer files from single connector
+    # in parallel. We could also transfer files belonging to different
+    # connectors in parallel but this could produce huge number of threads,
+    # since S3 uses multiple threads to transfer single file.
+    objects_to_transfer: dict[str, list[dict[str, str]]] = {}
+    for base_url, (connector_name, files) in response.message_data.items():
+        for file in files:
+            file["from_base_url"] = base_url
+            file["to_base_url"] = base_url
+        objects_to_transfer[connector_name] = (
```
Comment (dblenkus) [inline resolwe/flow/executors/init_container.py:84]: It would be nice to use `defaultdict` here.

File: resolwe/flow/executors/init_container.py [hunk 2]
```diff
@@ -86,23 +105,27 @@ async def transfer_inputs(communicator: BaseCommunicator):
...
-        t = Transfer(from_connector, to_connector)
-        t.transfer_objects(base_url, files)
-    except DataTransferError:
-        logger.exception(
-            "Data transfer error downloading data to base url {}.".format(base_url)
+    logger.info(f"Transfering data {from_connector} --> {to_connector}.")
+    transfer = Transfer(from_connector, to_connector)
+    # Start futures and evaluate their results. If exception occured it will
+    # be re-raised.
+    (
```
Comment (dblenkus) [inline resolwe/flow/executors/init_container.py:121]: I don't like list comprehension here, as it only makes code longer and more complex.

File: resolwe/storage/connectors/transfer.py [hunk 3]
```diff
@@ -146,9 +146,13 @@ def transfer_objects(
 
         return None if objects_stored is objects else objects_stored
 
-    def _transfer_chunk(self, url: Path, objects: Iterable[dict]) -> bool:
+    def transfer_chunk(self, url: Path, objects: Iterable[dict]) -> bool:
```
Comment (dblenkus) [inline resolwe/storage/connectors/transfer.py:149]: Commit message here is a bit misleading, as the commit only allows overriding of variables and has nothing to do with parallelization of upload (at least not directly).

File: resolwe/flow/executors/init_container.py (hunk 1 above)
Comment (gregorjerse) [reply resolwe/flow/executors/init_container.py:84]: Sure.

File: resolwe/flow/executors/init_container.py (hunk 2 above)
Comment (gregorjerse) [reply resolwe/flow/executors/init_container.py:121]: OK.

**Review by dblenkus — APPROVED** [review]

**Comment by gregorjerse** [comment]
I tested the PR on 'qa' and added some improvements based on the results.

**Review by dblenkus — APPROVED** [review]

**Pull request merged**